[enhance] skip docker build in hack/make.sh

### Ⅰ. Issue Description
We may need to build a test image and push it to registry and use it directly in test.
As sometimes, CI may fail in `docker build` step as following:

```
# cd .; git clone https://go.googlesource.com/tools /go/src/golang.org/x/tools
Cloning into '/go/src/golang.org/x/tools'...
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
package golang.org/x/tools/go/ast/astutil: exit status 128
package golang.org/x/tools/go/gcexportdata: cannot find package "golang.org/x/tools/go/gcexportdata" in any of:
	/usr/local/go/src/golang.org/x/tools/go/gcexportdata (from $GOROOT)
	/go/src/golang.org/x/tools/go/gcexportdata (from $GOPATH)
The command '/bin/sh -c go get -u golang.org/x/lint/golint' returned a non-zero code: 1
make: *** [integration-test] Error 1
```

### Ⅱ. Describe what happened


### Ⅲ. Describe what you expected to happen


### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.
2.
3.

### Ⅴ. Anything else we need to know?


### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/hack/make.sh
+++ b/hack/make.sh
@@ -1,18 +1,21 @@
 #!/usr/bin/env bash
 set -ex
 
-# This script is to build pouch binaries and execute pouch tests.
-
+#
+# This script is used to build pouch binaries and execute pouch tests.
+#
 TMP=$(mktemp -d /tmp/pouch.XXXXXX)
+DIR="$( cd "$( dirname "$0" )/.." && pwd )"
+cd $DIR/
+
 CONTAINERD_VERSION=
 RUNC_VERSION=
 NSENTER_VERSION=
 DUMB_INIT_VERSION=
 
-DIR="$( cd "$( dirname "$0" )/.." && pwd )"
-cd $DIR/
 SOURCEDIR=/go/src/github.com/alibaba/pouch
-IMAGE=pouch:test
+
+IMAGE="registry.hub.docker.com/letty/pouchci:latest"
 if [[ $SOURCEDIR != $DIR ]];then
 	[ -d $SOURCEDIR ] && rm -rf $SOURCEDIR
 	POUCHTOPDIR=$(dirname $SOURCEDIR)
@@ -20,35 +23,35 @@ if [[ $SOURCEDIR != $DIR ]];then
 	ln -sf $DIR/ $SOURCEDIR
 fi
 
-function get_containerd_version()
+function get_containerd_version
 {
 	if which containerd &>/dev/null; then
 		CONTAINERD_VERSION=$(containerd -v|cut -d " " -f 3)
 	fi
 }
 
-function get_runc_version()
+function get_runc_version
 {
 	if which runc &>/dev/null; then
 		RUNC_VERSION=$(runc -v|head -1| cut -d " " -f 3)
 	fi
 }
 
-function get_nsenter_version()
+function get_nsenter_version
 {
 	if which nsenter &>/dev/null; then
 		NSENTER_VERSION=$(nsenter -V | cut -d " " -f 4)
 	fi	
 }
 
-function get_dumb_init_version()
+function get_dumb_init_version
 {
 	if which dumb-init &>/dev/null; then
 		DUMB_INIT_VERSION=$(dumb-init -V 2>&1 | cut -d " " -f 2)
 	fi	
 }
 
-function install_containerd()
+function install_containerd
 {
 	echo "Try installing containerd"
 	get_containerd_version
@@ -56,12 +59,14 @@ function install_containerd()
 		echo "Containerd already installed."
 	else
 		echo "Download and install containerd."
-		wget --quiet https://github.com/containerd/containerd/releases/download/v1.0.3/containerd-1.0.3.linux-amd64.tar.gz -P $TMP
-		tar xf $TMP/containerd-1.0.3.linux-amd64.tar.gz -C $TMP && cp -f $TMP/bin/* /usr/local/bin/
+		wget --quiet \
+			https://github.com/containerd/containerd/releases/download/v1.0.3/containerd-1.0.3.linux-amd64.tar.gz -P $TMP
+		tar xf $TMP/containerd-1.0.3.linux-amd64.tar.gz -C $TMP &&
+			cp -f $TMP/bin/* /usr/local/bin/
 	fi;
 }
 
-function install_runc()
+function install_runc
 {
 	echo "Try installing runc"
 	get_runc_version
@@ -69,13 +74,14 @@ function install_runc()
 		echo "Runc already installed."
 	else
 		echo "Download and install runc."
-		wget --quiet https://github.com/alibaba/runc/releases/download/v1.0.0-rc4-1/runc.amd64 -P /usr/local/bin
+		wget --quiet \
+			https://github.com/alibaba/runc/releases/download/v1.0.0-rc4-1/runc.amd64 -P /usr/local/bin
 		chmod +x /usr/local/bin/runc.amd64
 		mv /usr/local/bin/runc.amd64 /usr/local/bin/runc
 	fi;
 }
 
-function install_lxcfs()
+function install_lxcfs
 {
 	echo "Try installing lxcfs"
 	if grep -qi "ubuntu" /etc/issue ; then
@@ -90,7 +96,7 @@ function install_lxcfs()
 	fi
 }
 
-function install_nsenter()
+function install_nsenter
 {
 	echo "Try installing nsenter"
 	get_nsenter_version
@@ -99,9 +105,24 @@ function install_nsenter()
 			echo "Nsenter already installed."
 		else
 			echo "Download and install nsenter."
-			apt-get -y install libncurses5-dev libslang2-dev gettext zlib1g-dev libselinux1-dev debhelper lsb-release pkg-config po-debconf autoconf automake autopoint libtool
-			wget --quiet https://www.kernel.org/pub/linux/utils/util-linux/v2.24/util-linux-2.24.1.tar.gz -P $TMP
-			tar xf $TMP/util-linux-2.24.1.tar.gz -C $TMP && cd $TMP/util-linux-2.24.1
+			apt-get -y install \
+				libncurses5-dev \
+				libslang2-dev \
+				gettext \
+				zlib1g-dev \
+				libselinux1-dev \
+				debhelper \
+				lsb-release \
+				pkg-config \
+				po-debconf \
+				autoconf \
+				automake \
+				autopoint \
+				libtool
+			wget --quiet \
+				https://www.kernel.org/pub/linux/utils/util-linux/v2.24/util-linux-2.24.1.tar.gz -P $TMP
+			tar xf $TMP/util-linux-2.24.1.tar.gz -C $TMP && 
+				cd $TMP/util-linux-2.24.1
 			./autogen.sh
 			autoreconf -vfi
 			./configure && make 
@@ -111,7 +132,6 @@ function install_nsenter()
 	else
 		yum install -y util-linux
 	fi
-
 }
 
 # Install dumb-init by downloading the binary.
@@ -123,42 +143,41 @@ function install_dumb_init
 		echo "Dumb-init already installed."
 	else
 		echo "Download and install dumb-init."
-		wget --quiet -O /tmp/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.1/dumb-init_1.2.1_amd64
+		wget --quiet -O /tmp/dumb-init \
+			 https://github.com/Yelp/dumb-init/releases/download/v1.2.1/dumb-init_1.2.1_amd64
 		mv tmp/dumb-init /usr/bin/
 		chmod +x /usr/bin/dumb-init
 	fi
 }
 
 # install pouch and essential binaries: containerd, runc and so on
-function install_pouch ()
+function install_pouch 
 {
-	# install containerd
 	install_containerd
-	# install runc
 	install_runc
 	# copy pouch daemon and pouch cli to PATH
 	echo "Install pouch."
 	cp -f $DIR/pouch $DIR/pouchd /usr/local/bin/
-	# install lxcfs
 	install_lxcfs
-	# install nsenter
 	install_nsenter
 }
 
-function target()
+function target
 {
 	case $1 in
 	check)
 		docker run --rm -v $(pwd):$SOURCEDIR $IMAGE bash -c "make check"
 		;;
 	build)
-		docker run --rm -v $(pwd):$SOURCEDIR $IMAGE bash -c "make build"  >$TMP/build.log ||
-		    { echo "make build log:"; cat $TMP/build.log; return 1; }
+		docker run --rm -v $(pwd):$SOURCEDIR $IMAGE \
+			bash -c "make build"  >$TMP/build.log ||
+			{ echo "make build log:"; cat $TMP/build.log; return 1; }
 		install_pouch  >$TMP/install.log ||
-		    { echo "install pouch log:"; cat $TMP/install.log; return 1; }
+			{ echo "install pouch log:"; cat $TMP/install.log; return 1; }
 		;;
 	unit-test)
-		docker run --rm -v $(pwd):$SOURCEDIR $IMAGE bash -c "make unit-test"
+		docker run --rm -v $(pwd):$SOURCEDIR $IMAGE \
+			bash -c "make unit-test"
 		;;
 	cri-test)
 		cd $SOURCEDIR
@@ -166,13 +185,19 @@ function target()
 		;;
 	integration-test)
 
-	    install_dumb_init || echo "Warning: dumb-init install failed! rich container related tests will be skipped"
-		docker run --rm -v $(pwd):$SOURCEDIR -e GOPATH=/go:$SOURCEDIR/extra/libnetwork/Godeps/_workspace $IMAGE bash -c "cd test && go test -c -o integration-test"
+		install_dumb_init ||
+			echo "Warning: dumb-init install failed!\
+				 rich container related tests will be skipped"
+		docker run --rm -v $(pwd):$SOURCEDIR \
+			-e GOPATH=/go:$SOURCEDIR/extra/libnetwork/Godeps/_workspace \
+			$IMAGE \
+			bash -c "cd test && go test -c -o integration-test"
 
 		#start pouch daemon
 		echo "start pouch daemon"
 		if stat /usr/bin/lxcfs ; then
-			pouchd --debug --enable-lxcfs=true --lxcfs=/usr/bin/lxcfs > $TMP/log 2>&1 &
+			pouchd --debug --enable-lxcfs=true \
+				--lxcfs=/usr/bin/lxcfs > $TMP/log 2>&1 &
 		else
 			pouchd --debug > $TMP/log 2>&1 &
 		fi
@@ -194,16 +219,15 @@ function target()
 			fi
 		done
 
-		pouch pull registry.hub.docker.com/library/busybox:1.28 >/dev/null
-
 		echo "verify pouch version"
 		pouch version
 
-        # copy tls file
-        cp -rf $DIR/test/tls /tmp/
+		# copy tls file
+		cp -rf $DIR/test/tls /tmp/
 
 		# If test is failed, print pouch daemon log.
-		$DIR/test/integration-test -test.v -check.v || { echo "pouch daemon log:"; cat $TMP/log; return 1; } 
+		$DIR/test/integration-test -test.v -check.v ||
+			{ echo "pouch daemon log:"; cat $TMP/log; return 1; } 
 		;;
 	*)
 		echo "no such target: $target"
@@ -213,9 +237,13 @@ function target()
 }
 
 
-function main ()
+function main 
 {
-	docker build --quiet -t $IMAGE . 
+	docker pull $IMAGE
+	if (( $? != 0 )); then
+		echo "ERR: pull $IMAGE failed"
+		exit 1
+	fi
 
 	if [[ $# < 1 ]]; then
 		targets="check build unit-test integration-test"

